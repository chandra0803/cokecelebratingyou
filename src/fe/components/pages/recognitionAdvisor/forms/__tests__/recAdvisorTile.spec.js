/*global jest*/

jest.mock( 'react-redux' );
import React from 'react';
import RecAdvisorTile from '../recAdvisorTile.js';
import renderer from 'react-test-renderer';

window.recognitionAdvisor = {
  raUrl: 'ra/reminders.action',
  raTilePageDisplay: 'yes',
  raDetailPageDisplay: 'no',
  raEndModelPageDisplay: 'no',
  content: []
};

const cmStringsArrayToObject = ( propertiesArray ) => {
    const propertiesObject = {};
    if( propertiesArray ) {

        propertiesArray.forEach( property => {
            propertiesObject[ `${ property.code }.${ property.key.toLowerCase() }` ] = property.content;
        } );
    }
    return propertiesObject;
};

const defaultProps = {
    recognitionAdvisor: {
      raUrl: 'ra/reminders.action',
      raTilePageDisplay: 'yes',
      raDetailPageDisplay: 'no',
      raEndModelPageDisplay: 'no',
      content: []
    }
};

const raTileModule = {
  'rowNumStart': 1,
  'rowNumEnd': 7,
  'activePage': 0,
  'sortColName': 'NO_VALUE',
  'sortedBy': '',
  'excludeUpcoming': 0,
  'filterValue': '',
  'pendingStatus': 1
};

it( 'renders the dom like the snapshot when the props have default values.', () => {
    const props = {
        ...defaultProps,
        recognitionAdvisorFunc: jest.fn(),
        raEligibleProgramsFunc: jest.fn()
    };
    const state = {
      attributeData: raTileModule,
      raTilePage: true,
      raDetailPage: false,
      raEndModelPage: false,
      content: cmStringsArrayToObject( window.recognitionAdvisor.content )
    };
    const component = renderer.create(
        <div>
            <RecAdvisorTile { ...state } { ...props } />
        </div>
    );

    const tree = component.toJSON();
    expect( tree ).toMatchSnapshot();
} );
