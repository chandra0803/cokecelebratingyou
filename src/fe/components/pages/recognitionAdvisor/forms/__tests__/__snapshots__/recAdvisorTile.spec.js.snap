// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the dom like the snapshot when the props have default values. 1`] = `
<div>
  <div>
    <h3
      className="module-title"
    />
    <div
      className="ra-wrapper"
    >
      <div
        className="clearfix list-container col-count-2"
      />
    </div>
    <div
      className="module-actions"
    >
      <a
        href="undefined/raDetails.do"
      />
    </div>
  </div>
</div>
`;
